src/modules/udw/services/universal.discovery.service.js

```diff
@@ -0,0 +1,243 @@
+import { showErrorNotification } from '../../common/services/notification.service';
+import { handleRequestResponse } from '../../common/helpers/request.helper.js';
+
+const HEADERS_CREATE_VIEW = {
+    Accept: 'application/vnd.ez.api.View+json; version=1.1',
+    'Content-Type': 'application/vnd.ez.api.ViewInput+json; version=1.1',
+};
+export const QUERY_LIMIT = 50;
+const ENDPOINT_CREATE_VIEW = '/api/ezp/v2/views';
+
+/**
+ * Loads preselected location data
+ *
+ * @function loadPreselectedLocationData
+ * @param {Object} params
+ * @param {String} params.startingLocationId
+ * @param {String} params.locationId
+ * @param {Number} [params.limit]
+ * @param {Function} callback
+ */
+export const loadPreselectedLocationData = ({ startingLocationId, locationId, limit = QUERY_LIMIT }, callback) => {
+    const endpoint = window.Routing.generate('ezplatform.udw.preselected_location.data', {
+        startingLocationId,
+        locationId,
+        limit,
+    });
+    const request = new Request(endpoint, {
+        mode: 'same-origin',
+        credentials: 'same-origin',
+    });
+
+    fetch(request)
+        .then(handleRequestResponse)
+        .then(callback)
+        .catch(showErrorNotification);
+};
+
+/**
+ * Loads content info
+ *
+ * @function loadContentInfo
+ * @param {Object} restInfo REST config hash containing: token and siteaccess properties
+ * @param {String} contentId
+ * @param {Function} callback
+ */
+export const loadContentInfo = ({ token, siteaccess }, contentId, callback) => {
+    const body = JSON.stringify({
+        ViewInput: {
+            identifier: `udw-load-content-info-${contentId}`,
+            public: false,
+            ContentQuery: {
+                Criteria: {},
+                FacetBuilders: {},
+                SortClauses: {},
+                Filter: { ContentIdCriterion: `${contentId}` },
+                limit: 1,
+                offset: 0,
+            },
+        },
+    });
+    const request = new Request(ENDPOINT_CREATE_VIEW, {
+        method: 'POST',
+        headers: { ...HEADERS_CREATE_VIEW, 'X-Siteaccess': siteaccess, 'X-CSRF-Token': token },
+        body,
+        mode: 'same-origin',
+        credentials: 'same-origin',
+    });
+
+    fetch(request)
+        .then(handleRequestResponse)
+        .then(callback)
+        .catch(showErrorNotification);
+};
+
+/**
+ * Loads location
+ *
+ * @function loadLocation
+ * @param {Object} params params hash containing REST config: token and siteaccess properties; locationId and offset
+ * @param {Function} callback
+ */
+export const loadLocation = ({ token, siteaccess, locationId, limit = QUERY_LIMIT, offset = 0 }, callback) => {
```

Comment:
As far as I remember this endpoint can load multiple locations when you pass location comma separated. So probably we should name this `loadLocations`

Reply:
I will correct it. It was just a copy-paste from the `src/modules/universal-discovery` directory